pytket/extensions/qiskit/backends/aer.py

```diff
@@ -188,48 +188,67 @@ def _arch_dependent_default_compilation_pass(
                     *arch_specific_passes,
                     SynthesiseTket(),
                 ],
-                False,
+            )
+        if optimisation_level == 2:
+            return SequencePass(
+                [
+                    DecomposeBoxes(),
+                    FullPeepholeOptimise(),
+                    *arch_specific_passes,
+                    CliffordSimp(False),
+                    SynthesiseTket(),
+                ],
             )
         return SequencePass(
             [
                 DecomposeBoxes(),
-                FullPeepholeOptimise(),
+                RemoveBarriers(),
```

Comment:
Why are you removing the barriers?

Reply:
`GreedyPauliSimp` doesn't support them - I'll update the documentation to make it clear that this optimisation level will remove them